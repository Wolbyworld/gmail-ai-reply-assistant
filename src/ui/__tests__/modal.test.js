import { describe, it, expect } from '@jest/globals';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

// Helper to get the directory name in ES Module scope
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Resolve the path to the HTML file relative to the test file
const modalHTMLPath = path.resolve(__dirname, '../modal.html');

describe('Modal HTML Structure', () => {
  it('should match the snapshot', () => {
    const modalHTMLContent = fs.readFileSync(modalHTMLPath, 'utf8');
    // Basic check to ensure file content was read
    expect(modalHTMLContent.length).toBeGreaterThan(0);
    // Snapshot the content
    expect(modalHTMLContent).toMatchSnapshot();
  });
}); 