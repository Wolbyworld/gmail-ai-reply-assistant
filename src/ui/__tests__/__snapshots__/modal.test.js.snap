// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal HTML Structure should match the snapshot 1`] = `
"<!DOCTYPE html>
<!-- Minimal HTML structure for the AI Reply Modal -->
<div class="ai-reply-overlay">
  <div class="ai-reply-modal" role="dialog" aria-modal="true" aria-labelledby="ai-modal-title">
    <h2 id="ai-modal-title">AI Reply Assistant</h2>
    <div class="ai-reply-content">
      <label for="ai-talking-points" class="visually-hidden">Enter bullet points or guidance for your reply</label>
      <textarea id="ai-talking-points" placeholder="Enter bullet points or guidance for your reply..." aria-label="Talking points for AI Reply"></textarea>
      <!-- Spinner will be added here dynamically later -->
      <div class="ai-reply-spinner" style="display: none;"></div>
    </div>
    <div class="ai-reply-actions">
      <button class="ai-reply-cancel">Cancel</button>
      <button class="ai-reply-submit">Submit</button>
    </div>
  </div>
</div> "
`;
